### PR TITLE
Restructure RAG doc to separate concepts from how-to

### PR DESCRIPTION
**Pull request by schnecle**

### CODE REVIEW

File: docs/rag.md [hunk 1]
```diff
@@ -151,33 +227,27 @@ export const indexPdf = defineFlow(
...
 
-To run the flow:
+#### Run the indexer flow
 
 ```posix-terminal
 genkit flow:run indexPdf "'../pdfs'"
 ```
 
-## Retrievers
+The local database now has documents stored in it!
```
Comment (kevinthecheung) [inline docs/rag.md:242]: The reader can't actually follow along since the PDFs don't exist for real. Maybe rephrase accordingly.

File: docs/rag.md [hunk 2]
```diff
@@ -111,6 +158,21 @@ configureGenkit({
...
+### Define an Indexer
+
+The following example shows how to create an indexer to ingest a collection of PDF documents
+and store them in a local vector database.
+
+It uses the local file-based vector similarity retriever
+that Genkit provides out-of-the box for simple testing and prototyping (_do not
+use in production_)
+
+#### Create the indexer and chunking config
```
Comment (kevinthecheung) [inline docs/rag.md:172]: Let's make the chunking thing its own section since it's an extra-Genkit step. Give it an intro like "This example uses the llm-chunk library to ... The following definition configures it to ..."

**Review by kevinthecheung — CHANGES_REQUESTED** [review]
Okay, yeah, way better. Thanks!

File: docs/rag.md [hunk 3]
```diff
@@ -78,21 +81,63 @@ with a stable source of data. On the other hand, if you are working with data
...
-The following example shows how you could ingest a collection of PDF documents
-into a vector database. It uses the local file-based vector similarity retriever
+### Embedders
+
+An embedder is a function that takes a collection of content and creates a numeric vector that is representative of the semantic meaning of the original text. As mentioned above, embedders are leveraged as part of the process of indexing, however, they can also be used independently to create embeddings independent of an index.
+
+### Retrievers
+
+A retriever is a concept that encapsulates logic related to any kind of document
+retrieval. The most popular retrieval cases typically include retrieval from
```
Comment (MaesterChestnut) [inline docs/rag.md:91]: nit; recommend changing `typically` to often in this line to avoid over using the word

**Review by MaesterChestnut — APPROVED** [review]

File: docs/rag.md (hunk 3 above)
Comment (schnecle) [reply docs/rag.md:91]: I'm noticing I say independently too much as well - will fix!

File: docs/rag.md (hunk 1 above)
Comment (schnecle) [reply docs/rag.md]: done.

File: docs/rag.md (hunk 2 above)
Comment (schnecle) [reply docs/rag.md:173]: done. WDYT?

File: docs/rag.md [hunk 4]
```diff
@@ -78,21 +81,63 @@ with a stable source of data. On the other hand, if you are working with data
...
-import { chunk } from 'llm-chunk'; // npm install llm-chunk
-import path from 'path';
-import pdf from 'pdf-parse'; // npm i pdf-parse && npm i -D --save @types/pdf-parse
-import z from 'zod';
+### Install dependencies for processing pdfs
 
-import { configureGenkit } from '@genkit-ai/core';
+```posix-terminal
+npm install llm-chunk pdf-parse
+npm i -D --save @types/pdf-parse
```
[the remark this answers is not in the document]
Comment (schnecle) [reply docs/rag.md]: done.

File: docs/rag.md [hunk 5]
```diff
@@ -78,21 +81,63 @@ with a stable source of data. On the other hand, if you are working with data
...
 
-```ts
-import { Document, index } from '@genkit-ai/ai/retriever';
-import { defineFlow, run } from '@genkit-ai/flow';
-import fs from 'fs';
-import { chunk } from 'llm-chunk'; // npm install llm-chunk
-import path from 'path';
-import pdf from 'pdf-parse'; // npm i pdf-parse && npm i -D --save @types/pdf-parse
-import z from 'zod';
+### Install dependencies for processing pdfs
```
[the remark this answers is not in the document]
Comment (schnecle) [reply docs/rag.md]: done.

File: docs/rag.md (hunk 3 above)
Comment (schnecle) [reply docs/rag.md:91]: done

File: docs/rag.md [hunk 6]
```diff
@@ -111,16 +159,55 @@ configureGenkit({
...
+### Define an Indexer
+
+The following example shows how to create an indexer to ingest a collection of PDF documents
+and store them in a local vector database.
+
+It uses the local file-based vector similarity retriever
+that Genkit provides out-of-the box for simple testing and prototyping (_do not
+use in production_)
+
+#### Create the indexer and chunking config
```
Comment (kevinthecheung) [inline docs/rag.md:173]:
```suggestion
#### Create the indexer
```

**Review by kevinthecheung — APPROVED** [review]
